libschrift: 0.9.1 -> 0.10.0

sternenseemann opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

<https://github.com/tomolt/libschrift/blob/master/CHANGELOG.md#v0100>

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

sternenseemann committed:
libschrift: 0.9.1 -> 0.10.0

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111152` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>libschrift: log <a href="https://termbin.com/dono">https://termbin.com/dono</a></li>
  </ul>
</details>

sternenseemann committed:
libschrift: fix build on darwin

On darwin unsigned long int != uint_fast32_t:
https://opensource.apple.com/source/Libc/Libc-1439.40.11/include/_types/_uint32_t.h.auto.html

Thus the build fails as SFT_Glyph is defined as unsigned long, but
schrift.c uses uint_fast32_t everywhere instead of SFT_Glyph. We fix
this by adjusting the int type in the header which should be fine for
now.

This patch is technically not required for Linux, but it doesn't hurt
either as unsigned long == uint_fast32_t.

sternenseemann commented:
@SuperSandro2000 can you test the latest commit on darwin for me? This should fix the build.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111152` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>libschrift</li>
</ul>
</details>

SuperSandro2000 commented on pkgs/development/libraries/libschrift/default.nix:
```diff
@@ -2,18 +2,22 @@
       --replace "PREFIX = /usr/local" "PREFIX = $out"
+  '' + lib.optionalString stdenv.isDarwin ''
+    # on darwin unsigned long int != uint_fast32_t
+    sed -i 's/typedef unsigned long\s\+SFT_Glyph;/#include <stdint.h>\
```
Should we use a patch for this? This could easily bit rot.

sternenseemann replied:
Now that the same change has been merged into upstream I could probably add the commit as a patch, but it's gonna have to be removed at the next release either way.

SuperSandro2000 commented on pkgs/development/libraries/libschrift/default.nix:
```diff
@@ -1,16 +1,22 @@
   };
 
+  # fix a compilation failure related to darwin integers, remove at next release
+  patches = lib.optional stdenv.isDarwin (fetchpatch {
```
Sorry for another roundtrip but could we apply this patch to all platforms? Otherwise you or someone else might miss removing it.

```suggestion
  patches = (fetchpatch {
```

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111152` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>libschrift</li>
</ul>
</details>
